docs/arch/adr-003-versioning-branching-strategy.md

```diff
@@ -0,0 +1,66 @@
+
+We need a branching strategy to support the goals of the versioning strategy. We do not want to have to stop development on `main` as we await a release, nor do we want every change in `main` published in every release. Using branches, we can decide which changes go into which release.
+
+## Requirements
+The following are requirements that must be met for releasing dbt:
+
+1. We must have major, minor, and patch releases.
+1. We must have release types that have the goal of garnering early feedback from the Community.
+1. We must have release types that aim to identify bugs and test for stability.
+1. We must have release types that are stable and of production quality.
+1. We must have the ability to isolate changes from different versions based on the type of release.
+
+## Decisions
+
+### Version Types
+<details>
+<summary> Header Explanations: </summary>
+
+- Type: the type of the version (e.g. beta, final)
+- Released?: does this type of version need to be released?
+- Branches: the branches where these version types are present (e.g. beta version numbers should only exist on the `main` branch)
+- Release Versions: the releases where the version type is applicable (i.e. we will only have alphas for major and minor releases, we will not have alphas for patch releases)
+- Expectations: the stability of the code changes in the release
+</details>
+
+| Type | Released? | Branches | Release Versions | Expectations |
+| ---- | --------- | -------- | ---------------- | ------------ |
+| Alpha | No | `main` | Major / Minor | Experimental |
+| Beta | Yes | `main` | Major / Minor | Experimental |
+| RC | Yes | Release branch | Major / Minor / Patch | Pre-production |
+| Final | Yes | Release branch | Major / Minor / Patch | Production |
+
+### Branching Strategy
+Based off of the release version and expectations, the version type's branching strategy can be determined. 
+ * Alpha and Beta versions are experimental, and therefore exist only on the `main` branch, where all our changes reside
+ * Alpha versions are never released. They signify that development is underway, though the changes they contain have not been released in any form
+ * An RC version denotes a more stable state. A release branch will be created for RC releases to limit the changes a release contains
+ * Final versions are stable and tested. Only verified changes will go into these releases, which are inherited from the corresponding RC version
+ * A release branch will exist for each unique major or minor version, to be named accordingly: `<major>.<minor>.latest` (e.g. `1.0.latest`)
+
+ ![Branching Strategy](images/ReleasingBranchStrategy.png)
```

Comment:
to this point, maybe this doc can explicitly describe what the plan is for a commit we do _not_ want to go out in the next release is?

Reply:
Historically, everything in `main` has gone into the release branch. We have never started (not saying this can't happen in the future) working on something that was slated for the next release before the release branch was cut for the upcoming release. In most cases, I would think this would most likely happen for a large feature and in this case we should leverage feature branches which I can add to the doc to make it clear on how we should handle something like this if it would arise

Reply:
yeah that works. just something we can have to start with and iterate on.

Reply:
I added a section about this so let me know if that helps clarify things

Reply:
it does! thanks.